apex/contrib/torchsched/utils.py

```diff
@@ -0,0 +1,246 @@
+"""General utilities for graph scheduler study."""
+
+from __future__ import annotations
+
+import math
+from typing import Callable
+
+import torch
+import torch.nn as nn
+
+__all__ = ["partition_weighted", "partition_even", "run_and_time"]
+
+cudart = torch.cuda.cudart()
+
+
+def partition_weighted(
+    weights: list[int] | tuple[int, ...],
+    sm_count: int | None = None,
+    rounding_multiple: int = 1,
+) -> list[int]:
+    """Partition the SMs based on their weights, rounding to a specified multiple.
+
+    This function distributes a total number of SMs proportionally to their weight counts,
```

Comment:
"distributes a total number of SMs" is somewhat awkward, is the meaning "map #GEMM weights to a number of SMs?"

Reply:
Removed the file, we can discuss online.